decorator.rb

```diff
@@ -0,0 +1,23 @@
+# Base Decorator class
+require_relative 'person'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+end
```

Comment:
- [ ] Please create a `correct_name` method that returns the result of the `correct_name` method of the `@namable` is implemented. Check below :point_down:  my suggestion  :slightly_smiling_face: 

```suggestion
class Decorator < Nameable
    attr_accessor :nameable
  
    def initialize(nameable)
      @nameable = nameable
    end
  

    def correct_name
      @nameable.correct_name
    end
end
```